README.rst

```diff
@@ -0,0 +1,32 @@
+ada-url
+========
+
+This is ``ada_url``, a Python library for parsing and joining URLs.
+
+Examples
+--------
+
+.. code-block:: python
+
+    >>> import ada_url
+    >>> ada_url.check_url('https://example.org')
+    True
+    >>> ada_url.join_url(
+        'https://example.org/dir/child.txt', '../parent.txt'
+    )
+    'https://example.org/parent.txt'
+    >>> ada_url.normalize_url('https://example.org/dir/../parent.txt')
+    'https://example.org/parent.txt'
+    >>> ada_url.parse_url('https://user:pass@example.org:80/api?q=1#2')
+    {
+        'href': 'https://user:pass@example.org:80/api?q=1#2',
+        'username': 'user',
+        'password': 'pass',
+        'protocol': 'https:',
+        'host': 'example.org:80',
+        'port': '80',
+        'hostname': 'example.org',
+        'pathname': '/api',
+        'search': '?q=1',
+        'hash': '#2'
+    }
```

Comment:
```suggestion
    }
 
 Building and testing from source
 ----------------------------------

Make sure that you have installed python 3 with development files. If you are under macOS and you have installed python through brew, you have already what is needed. Under Linux, you may need a command such as  "yum install python-devel" (redhat, rocky linux,...) or "apt-get install python-dev". 

You will also need a recent C++ toolchain installed. Under macOS, it is provided by Xcode. It is typically already available under Linux.

Ensure that you have pip installed: `python3 -m ensurepip --upgrade`.

Grab the package from GitHub:
    git clone https://github.com/ada-url/python/ ada_url_python
    cd ada_url_python

Observe that we check out the project in a different directory (`ada_url_python`) because the `python` name may clash with other projects, and is an unfortunate choice.

While in the ada_url python package directory, enter into a virtual environment:

    python3 -m pip install --user virtualenv
    python3 -m venv env
    source env/bin/activate

Install the requirements, build ada.o, and run the python build...

    python -m pip install -r requirements/development.txt
    c++ -c "ada_url/ada.cpp" -fPIC -std="c++17" -O2 -o "ada_url/ada.o"
    python -m build --no-isolation

You can now enter the python environment and call ada_url. However, if you remain within the python package directory, python might get confused by the existence of the `ada_url` directory, which, unfortunately, does not contain the build package and clashes with the python package name. However, you can fix that temporarily moving the directory...

    mv ada_url tmp

You should not forget to move it back once you are done.

Then you can launch the python interpreter:

    >>> u = ada_url.parse_url("http://www.Goolgle.com/bah")
    >>> u["href"] = "http://www.banana.com"
    >>> u
    {'href': 'http://www.banana.com', 'username': '', 'password': '', 'protocol': 'http:', 'port': '', 'hostname': 'www.goolgle.com', 'host': 'www.goolgle.com', 'pathname': '/bah', 'search': '', 'hash': '', 'origin': 'http://www.goolgle.com'}

As this example illustrates, the result from parsing is a object containing independent values and is not a WHATWG URL compliant instance. 

You can also run unit tests:
   python -m unittest

To exit the virtual environment, once you are done type `deactivate`.
```

Reply:
> Observe that we check out the project in a different directory (`ada_url_python`) because the `python` name may clash with other projects, and is an unfortunate choice.

Should we fix this unfortunate choice before bringing in this PR? (I have no opinion either way)

Reply:
> You can now enter the python environment and call ada_url. However, if you remain within the python package directory, python might get confused by the existence of the `ada_url` directory, which, unfortunately, does not contain the build package and clashes with the python package name. However, you can fix that temporarily moving the directory...

Was this actually a problem?

Reply:
> Should we fix this unfortunate choice before bringing in this PR?

It is an issue. At a minimum, we need to document it.

Reply:
I've written some instructions that I think should work. You can see them in the `docs/index.rst`, but here are the steps:
* Don't change directories from the root
* Run the `make package` command
* `pip install -e .` to do a "local install"
* Now tests will succeed and you can import from the REPL

This is fairly common in the Python world, from what I have seen.